client/lib/plugins/sanitize-section-content.js

```diff
@@ -126,6 +126,10 @@ const replacementFor = ( node ) => {
  * @returns {string} sanitized HTML
  */
 export const sanitizeSectionContent = ( content ) => {
+	if ( 'undefined' === typeof DOMParser ) {
```

Comment:
I'm not convinced this is always the right level to disable browser-only behavior to enable SSR. 
For example, in this instance, are we just going to leave the content unsanitized, and potentially vulnerable?

Reply:
@tyxla I pushed a [workaround](https://github.com/Automattic/wp-calypso/pull/66344/commits/95b6357cd408ad7525844407ba384c3773fd2453#diff-33ff16937e1a3218ed65c949cc79ec6f88ec3efb06d89d8d1cb1e0f0f49b32ae) based on [htmlparser2](https://github.com/fb55/htmlparser2), [domutils](https://github.com/fb55/domutils) and [dom-serializer](https://github.com/cheeriojs/dom-serializer)

These should load only on the server and replaces the `window` APIs used in content sanitization with server-friendly equivalents.

Reply:
Thanks for this! I haven't reviewed thoroughly, but given the complexity that the additional boilerplate provides, I wonder if we've tried a more targeted and ready-to-use solution, like [sanitize-html](https://www.npmjs.com/package/sanitize-html) for example?

Reply:
`sanitize-html` seems to have been used previously but it was replaced in #19316 with this `sanitizer-section-content`. 
Since then, a few other transforms have been added and the outputs have diverge. Running the `sanitizer-section-content` tests against the `sanitize-html`  results in most tests failing.
 
Even if we configure `sanitize-html` and `sanitizer-section-content` to return similar outputs, I think it'll become trickier to maintain and develop both approaches.

Reply:
Okay, I see. I know we're going in circles, but are you sure that sanitizing this on the endpoints won't be easier than what's being suggested here? I understand they're 4 endpoints, but it sounds like we could reuse the functionality.

Reply:
I don't think it'd be any easier because one of the endpoints is on `api.wordpress.org`. 
I'm not too familiar with that codebase I doubt we'll be able to push the changes. Even so, we'd probably need to implement and then manage the content sanitization on both platforms.

Reply:
Thanks for confirming. That does add additional challenges, indeed. 

Let's take what you have as a solution. What remains to be done in order for this PR to be ready for review? I see some tests are failing.

Reply:
The tests should be addressed: the unit tests needed the magnificent locale mocked, and the e2e was caused by the transition between SSR and client render defocusing the search input.

I think there should be enough here ready for review. 
From my testing, the routes work as expected and the content renders properly on both mediums. 

There's still some flickering mostly caused by a few components rendering differently based on breakpoint but it might not be worth addressing them in this already massive PR.